src/statemachine/statemachine_coroutine.js

```diff
@@ -0,0 +1,225 @@
+pc.extend(pc, (function () {
```

Comment:
Agreed. You’ll have to point me in the direction of how to do that update management - is there somewhere it already happens?

Also on reflection, I’m going to move all of the curve and curveset stuff to interpolation.  Its as good a home as any and keeps everything else pretty clean.


> On 22 Jun 2015, at 12:58, David Evans <notifications@github.com> wrote:
> 
> In src/statemachine/statemachine_coroutine.js <https://github.com/playcanvas/engine/pull/326#discussion_r32926757>:
> 
> > @@ -0,0 +1,225 @@
> > +pc.extend(pc, (function () {
> We could probably move co-routine into core_coroutine.js. Especially if we remove the dependency on the component system.
> 
> This is broader than just state machines.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/playcanvas/engine/pull/326/files#r32926757>.
> 


-- 
See our 2 minute video on DBS bringing gamification to Singapore here: 
www.3radical.com/resources/videos/#DBSVideo


3radical Limited, 1 Friary, Temple Quay, Bristol BS1 6EA, England.  
T: +44 117 344 5008 ¦ F : +44 117 344 5188

3radical Pte Ltd, 3 Church Street, Level 8 Samsung Hub, Singapore 049483.  
T: +65 6408 3711 ¦ F : +65 6408 0101

www.3radical.com

3radical Limited is a limited company registered in England and Wales 
under number 7872556.  The registered address of 3radical Limited is 1 
Friary, Temple Quay, Bristol BS1 6EA, England.

3radical Pte Ltd is a registered in Singapore under number 201312239K.  The 
registered address of 3radical Pte Ltd is 51 Goldhill Plaza, #07-10/11, 
Singapore 308900.

The information in this e-mail and any attachments may be confidential.  If 
you are not the intended recipient, please notify the sender as soon as 
possible and delete any copies. Unauthorised use, dissemination, 
distribution, publication or copying of this communication is prohibited.

Attachments to this email have been checked with virus detection 
software before transmission, however you should carry out virus checks 
before opening any attachment. 3radical Limited and 3radical Pte Ltd 
accepts no liability for any loss or damage which may be caused by software 
viruses.